### ISSUE TITLE
Console reported job status dead when job is actually alive

### ISSUE THREAD

**Issue by davin** [discoproject/disco#115]
It seems that computers do lie after all. :)  I have a script with a loop containing a long sequence of chained jobs.  After running a few to dozen minutes on my dual core lap-cluster, the same script would crash with either issue #104 (Invalid HTTP reply (expected 200 got 404)) or the error:

<code>
  File "/home/disco/disco/pydisco/disco/core.py", line 172, in wait
    return self.check_results(name, start_time, poll_interval \* 1000)
  File "/home/disco/disco/pydisco/disco/core.py", line 161, in check_results
    raise JobError("Job status %s" % status, self.host, name)
JobError: Job http://localhost:8989/myJob@4f3:7fe0f:b7d07 failed: Job status dead
</code>

However, on the status html page for myJob@4f3:7fe0f:b7d07, it shows that the dead job is in fact finished:

READY: Job finished in 0:00:02.044

I thought this might be related to comm_curl.py so I increased MAX_RETRIES, CONNECT_TIMEOUT and TRANSFER_TIMEOUT to no avail.

**Comment by davin** [comment]
Just to make sure, I also tracked the job down to /srv/disco/data/_disco_4444/19/myJob@4f3:7fe0f:b7d07.  The events file shows that the job is finished.

["2010/03/02 11:56:31","master","New job initialized!"]
["2010/03/02 11:56:31","master","Starting job"]
["2010/03/02 11:56:31","master","Map phase"]
["2010/03/02 11:56:31","master","map:0 added to waitlist"]
["2010/03/02 11:56:31","master","map:0 assigned to localhost"]
["2010/03/02 11:56:31","master","map:1 added to waitlist"]
["2010/03/02 11:56:31","master","map:1 assigned to localhost"]
["2010/03/02 11:56:31","master","map:2 added to waitlist"]
["2010/03/02 11:56:31","master","map:3 added to waitlist"]
["2010/03/02 11:56:32","localhost","[map:0] Received a new map job!"]
["2010/03/02 11:56:32","localhost","[map:0] Done: 50 entries mapped in total"]
["2010/03/02 11:56:32","localhost","[map:0] Task finished in 0:00:00.556"]
["2010/03/02 11:56:32","master","Received results from map:0 @ localhost."]
["2010/03/02 11:56:32","master","map:3 assigned to localhost"]
["2010/03/02 11:56:32","localhost","[map:3] Received a new map job!"]
["2010/03/02 11:56:32","localhost","[map:3] Done: 50 entries mapped in total"]
["2010/03/02 11:56:32","localhost","[map:3] Task finished in 0:00:00.180"]
["2010/03/02 11:56:32","master","Received results from map:3 @ localhost."]
["2010/03/02 11:56:32","master","map:2 assigned to localhost"]
["2010/03/02 11:56:32","localhost","[map:2] Received a new map job!"]
["2010/03/02 11:56:32","localhost","[map:2] Done: 50 entries mapped in total"]
["2010/03/02 11:56:32","localhost","[map:2] Task finished in 0:00:00.196"]
["2010/03/02 11:56:32","master","Received results from map:2 @ localhost."]
["2010/03/02 11:56:33","localhost","[map:1] Received a new map job!"]
["2010/03/02 11:56:33","localhost","[map:1] Done: 50 entries mapped in total"]
["2010/03/02 11:56:33","localhost","[map:1] Task finished in 0:00:01.365"]
["2010/03/02 11:56:33","master","Received results from map:1 @ localhost."]
["2010/03/02 11:56:33","master","Map phase done"]
["2010/03/02 11:56:33","master","Starting reduce phase"]
["2010/03/02 11:56:33","master","reduce:0 added to waitlist"]
["2010/03/02 11:56:33","master","reduce:0 assigned to localhost"]
["2010/03/02 11:56:33","master","reduce:1 added to waitlist"]
["2010/03/02 11:56:33","master","reduce:1 assigned to localhost"]
["2010/03/02 11:56:33","localhost","[reduce:0] Received a new reduce job!"]
["2010/03/02 11:56:33","localhost","[reduce:1] Received a new reduce job!"]
["2010/03/02 11:56:33","localhost","[reduce:0] Starting reduce"]
["2010/03/02 11:56:33","localhost","[reduce:1] Starting reduce"]
["2010/03/02 11:56:33","localhost","[reduce:0] Reduce done: 100 entries reduced in total"]
["2010/03/02 11:56:33","localhost","[reduce:0] Reduce done"]
["2010/03/02 11:56:33","localhost","[reduce:0] Task finished in 0:00:00.553"]
["2010/03/02 11:56:33","master","Received results from reduce:0 @ localhost."]
["2010/03/02 11:56:33","localhost","[reduce:1] Reduce done: 100 entries reduced in total"]
["2010/03/02 11:56:33","localhost","[reduce:1] Reduce done"]
["2010/03/02 11:56:33","localhost","[reduce:1] Task finished in 0:00:00.575"]
["2010/03/02 11:56:33","master","Received results from reduce:1 @ localhost."]
["2010/03/02 11:56:33","master","Reduce phase done"]
["2010/03/02 11:56:33","master","READY: Job finished in 0:00:02.044"]

**Comment by davin** [comment]
I also just tried increasing timeout in wait() to 6 (secs ?) but I still get either issue #104 or this one.

**Comment by davin** [comment]
I think I found the culprit.  It may have to do with poll_interval after all.  By setting poll_interval to 10 (secs ?) in wait(), increasing MAX_RETRIES, CONNECT_TIMEOUT and TRANSFER_TIMEOUT in comm_curl.py, I was able to go a lot further than before with my script.  No issue #104 or this one so far.  I'll try later with just changing poll_interval and not comm_curl.py to see if poll_interval is the root cause.
